src/server/game/AI/CreatureAI.cpp

```diff
@@ -28,11 +28,13 @@
 #include "CreatureTextMgr.h"
 
 //Disable CreatureAI when charmed
-void CreatureAI::OnCharmed(bool /*apply*/)
+void CreatureAI::OnCharmed(bool apply)
 {
-    //me->IsAIEnabled = !apply;*/
-    me->NeedChangeAI = true;
-    me->IsAIEnabled = false;
+    if (apply)
```

Comment:
If u add this, u will need all the other overrides, so better to have this just in one place :)